messaging/src/main/java/org/axonframework/eventhandling/annotation/AnnotatedEventHandlingComponent.java

```diff
@@ -135,7 +133,7 @@ private void registerHandler(MessageHandlingMember<? super T> handler) {
         handlingComponent.subscribe(
                 qualifiedName,
                 (event, ctx) ->
-                interceptorChain.handle(event, ctx, target, handler).ignoreEntries().cast()
+                interceptorChain.handle(event.withConvertedPayload(handler.payloadType(), converter), ctx, target, handler).ignoreEntries().cast()
```

Comment:
Can't the `Converter` come from the `ProcessingContext` here?

Reply:
Of course - done! :)